api/apps/api/src/modules/scenarios/locks/lock.service.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import { Either, left, right } from 'fp-ts/Either';
+
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import {
+  AcquireFailure,
+  lockedScenario,
+  lockNotFoundError,
+  LockService,
+  ReleaseFailure,
+} from './lock.service';
+import { ScenarioLockEntity } from './scenario.lock.entity';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+it(`should be able to grab lock if lock is available`, async () => {
+  await fixtures.GivenLockForScenarioIsNotFound();
+  const lock = await fixtures.WhenAcquiringALock();
+  await fixtures.ThenScenarioIsLocked(lock);
+});
+
+it(`should not be able to grab lock if lock is not available`, async () => {
+  await fixtures.GivenLockForScenarioIsFound();
+  const lock = await fixtures.WhenAcquiringALock();
+  await fixtures.ThenALockedScenarioErrorIsReturned(lock);
+});
+
+it(`should be able to release the lock if lock exists`, async () => {
+  await fixtures.GivenLockForScenarioIsFound();
+  const lock = await fixtures.WhenSameUserReleasesTheLock();
+  await fixtures.ThenScenarioIsNotLocked(lock);
+});
+
+it(`should not be able to release the lock if lock exists for a different user`, async () => {
+  await fixtures.GivenLockForScenarioIsNotFound();
+  const lock = await fixtures.WhenDifferentUserReleasesTheLock();
+  await fixtures.ThenScenarioIsStillLocked(lock);
+});
```

Comment:
I am not sure I get what we are testing here - reading the fixtures code, what happens makes sense (and indeed the test passes), but isn't `ThenScenarioIsStillLocked` misleading then? in this outcome, we don't really assert that the scenario is locked (and by extension that it is _still_ locked), if I am not missing something.

in any case, this is very minor - I may have misunderstood or it may just be that the description of one of the steps could be made more precise.

Reply:
ahah, yeah, the name is quite misleading, apologies for that :sweat_smile: I'll update the name, thanks for pointing out!!